### PR TITLE
Mimirtool: add the ability to include extra headers with requests made to the server

### LINKED ISSUE #7139: Mimirtool Rules: Add the ability to include extra headers with the request (for authentication or whatever else)
Within an environment there are cases where Mimir might be behind a reverse proxy. The only option for authentication is basic auth or an auth token. This may not work in all environments.

Cloudflare Service Tokens are a good example: https://developers.cloudflare.com/cloudflare-one/identity/service-tokens/ as they require Cloudflare custom headers.

Rather than provide a gazillion different authentication options, it would be easier and more flexible to include a generic option add extra headers to mimirtool when making rules requests.

### PR DESCRIPTION
**Pull request by snowzach**
This PR adds a `--extra-headers` CLI option and `MIMIR_EXTRA_HEADERS` environment variable option to the rules function of Mimir tool.

This allows to use various other authentication schemes as needed (Cloudflare Service Tokens as an example) without needing to modify mimirtool for every type.

Fixes #7139

#### Checklist

- [x] Tests updated.
- [X] Documentation added.
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [x] [`about-versioning.md`](/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: docs/sources/mimir/manage/tools/mimirtool.md [hunk 1]
```diff
@@ -187,6 +187,12 @@ The commands in this section enable you to perform the following actions:
 Some of the functionality that `mimirtool rules` offers is also available as a GitHub Action.
 For more information, refer to the [documentation of Mimirtool Github Action](https://github.com/grafana/mimir/blob/main/operations/mimir-rules-action/README.md).
 
+Configuration options relevant to rules commands:
+| Flag | Description |
+| ----------------- | ---------------------------------------------------------- |
+| `--auth-token` | Authentication token for bearer token or JWT auth. |
+| `--extra-headers` | Extra headers to add to requests for authentication |
```
Comment (jhesketh) [inline docs/sources/mimir/manage/tools/mimirtool.md:194]: Perhaps worth noting something along the lines of "in key=value format"?

**Review by jhesketh — APPROVED** [review]
Looks good to me. Will leave for somebody more familiar with mimirtool to merge.

File: operations/mimir-rules-action/README.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ This action is configured using environment variables defined in the workflow. T
...
 | `MIMIR_API_KEY`              | Optional password that is required for password-protected Mimir clusters. An encrypted [github secret](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets) is recommended. Used as the password under HTTP Basic authentication. | `false`  | N/A     |
 | `MIMIR_AUTH_TOKEN`           | Optional bearer or JWT token that is required for Mimir clusters authenticating by bearer or JWT token. An encrypted [github secret](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets) is recommended.                         | `false`  | N/A     |
+| `MIMIR_EXTRA_HEADERS`        | Newline-separated list of header=value pairs to add to the request.                                                                                                                                                                                                                                | `false`  | N/A     |
```
Comment (pracucci) [inline operations/mimir-rules-action/README.md:15]: In the description, I would clarify it's only supported by `mimirtool rules` command.

File: docs/sources/mimir/manage/tools/mimirtool.md [hunk 3]
```diff
@@ -187,6 +187,12 @@ The commands in this section enable you to perform the following actions:
 Some of the functionality that `mimirtool rules` offers is also available as a GitHub Action.
 For more information, refer to the [documentation of Mimirtool Github Action](https://github.com/grafana/mimir/blob/main/operations/mimir-rules-action/README.md).
 
+Configuration options relevant to rules commands:
```
Comment (pracucci) [inline docs/sources/mimir/manage/tools/mimirtool.md:190]: We need a newline here to get our doc correctly formatted:

```suggestion
Configuration options relevant to rules commands:

```

File: CHANGELOG.md [hunk 4]
```diff
@@ -104,6 +104,7 @@
 
 ### Mimirtool
 
+* [ENHANCEMENT] Add `--extra-headers` option to `mimirtool rules` command to add extra headers to requests for auth. #7139
```
Comment (pracucci) [inline CHANGELOG.md:107]: We're used to link the PR number:

```suggestion
* [ENHANCEMENT] Add `--extra-headers` option to `mimirtool rules` command to add extra headers to requests for auth. #7141
```

**Review by pracucci — APPROVED** [review]
I manually tested it, LGTM, thanks! I left few nits I would like to get fixed before merging.

File: operations/mimir-rules-action/README.md (hunk 2 above)
Comment (snowzach) [reply operations/mimir-rules-action/README.md]: Got it!

**Pull request merged**